ipn/ipnstate/ipnstate.go

```diff
@@ -112,28 +114,13 @@ type StatusBuilder struct {
 	st     Status
 }
 
-func (sb *StatusBuilder) SetVersion(v string) {
-	sb.mu.Lock()
-	defer sb.mu.Unlock()
-	sb.st.Version = v
-}
-
-func (sb *StatusBuilder) SetBackendState(v string) {
-	sb.mu.Lock()
-	defer sb.mu.Unlock()
-	sb.st.BackendState = v
-}
-
-func (sb *StatusBuilder) SetAuthURL(v string) {
+// MutateStatus calls f with the status to mutate.
+// It may not assume other fields of status are already populated, and
+// may not retain or write to the Status after f returns.
+func (sb *StatusBuilder) MutateStatus(f func(*Status)) {
```

Comment:
The passed in callbacks are called under a lock. It might be worth documenting what those callbacks can't do without a deadlock